src/components/TranslationHelpsWrapper.js

```diff
@@ -1,139 +1,110 @@
 // helpers
 import * as tHelpsHelpers from '../helpers/tHelpsHelpers';
 
-class TranslationHelpsWrapper extends React.Component {
-  constructor(props) {
-    super(props);
-    this.state = {
-      showHelpsModal: false,
-      modalArticle: '',
-      articleCategory: '',
-    };
+// resourcesReducer needs to be global so that the followTHelpsLink has the new article's content
+let resourcesReducer = {};
+
+function useTnArticleState(initialState) {
+  const [
+    {
+      showHelpsModal, modalArticle, articleCategory,
+    },
+    setTnArticleState,
+  ] = useState(initialState);
+
+  return {
+    showHelpsModal,
+    modalArticle,
+    articleCategory,
+    setThState: (updatedValues) => {
+      setTnArticleState(prevState => ({ ...prevState, ...updatedValues }));
+    },
+  };
+}
 
-    this.toggleHelpsModal = this.toggleHelpsModal.bind(this);
-    this.followTHelpsLink = this.followTHelpsLink.bind(this);
-    window.followLink = this.followTHelpsLink;
-  }
+function TranslationHelpsWrapper(props) {
+  const {
+    toolsSelectedGLs,
+    toolsReducer: { currentToolName },
+    contextIdReducer: { contextId },
+    showHelps,
+    toggleHelps,
+    translate,
+    actions,
+  } = props;
+  resourcesReducer = props.resourcesReducer;
+
+  const initialState = {
+    showHelpsModal: false,
+    modalArticle: '',
+    articleCategory: '',
+  };
+  const {
+    showHelpsModal,
+    modalArticle,
+    articleCategory,
+    setThState,
+  } = useTnArticleState(initialState);
+  const groupId = contextId.groupId;
+  const languageId = toolsSelectedGLs[currentToolName];
+
+  function followTHelpsLink(link) {
+    const linkParts = link.split('/'); // link format: <lang>/<resource>/<category>/<article>
 
-  componentDidMount() {
-    this.loadArticle(this.props);
-  }
+    const [lang, type, category, article] = linkParts;
+    const resourceDir = tHelpsHelpers.getResourceDirByType(type);
 
-  componentDidUpdate(prevProps) {
-    const { contextIdReducer } = this.props || {};
-    const prevContextIdReducer = prevProps.contextIdReducer;
+    actions.loadResourceArticle(resourceDir, article, lang, category);
+    const articleData = resourcesReducer.translationHelps[resourceDir][article];
 
-    if (this.getGroupId(contextIdReducer) !== this.getGroupId(prevContextIdReducer)) { // we only need to reload article when groupId changes
-      this.loadArticle(this.props);
-    }
+    setThState({
+      showHelpsModal: true,
+      modalArticle: articleData || `Cannot find an article for ${link}`,
+      articleCategory: category,
+    });
+    return true;
+  };
+  window.followLink = followTHelpsLink;
 
-    if (!isEqual(contextIdReducer, prevContextIdReducer)) { // we need to scroll to top whenever contextId changes
-      const page = document.getElementById('helpsbody');
+  useEffect(() => {
+    // if (groupId) { // may not be needed
+    actions.loadResourceArticle(currentToolName, groupId, languageId);
```

Comment:
@richmahn - have a suspicion that this may run slow, since it will reload the resource article every time the contextId changes.

Reply:
Nevermind, it's in dependency list.

Reply:
This does show that it only changes on groupId change, so not sure where you are seeing contextId.